From 59b53d4bc3ff25c0040e7ec6eb0f18288548f714 Mon Sep 17 00:00:00 2001
From: Yin Fengwei <fengwei.yin@intel.com>
Date: Wed, 10 Jan 2018 20:22:07 +0800
Subject: [PATCH 0915/2367] Klocwork: Fix Klocwork 21299 issue.

"'gpa2hpa.hpa' is used. But is uninitialized."

Change-Id: I4cee159dab8f118f3b4a60d02029dba618a91cac
Tracked-On:218445
Signed-off-by: Yin Fengwei <fengwei.yin@intel.com>
Reviewed-on: https://android.intel.com/614936
Reviewed-by: Chi, Mingqiang <mingqiang.chi@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/vhm/vhm_vm_mngt.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/vhm/vhm_vm_mngt.c b/drivers/vhm/vhm_vm_mngt.c
index fb47c12..100435d 100644
--- a/drivers/vhm/vhm_vm_mngt.c
+++ b/drivers/vhm/vhm_vm_mngt.c
@@ -136,6 +136,7 @@ unsigned long vhm_vm_gpa2hpa(unsigned long vmid, unsigned long gpa)
 	int ret;
 
 	gpa2hpa.gpa = gpa;
+	gpa2hpa.hpa = -1UL; /* Init value as invalid gpa */
 	ret = hcall_vm_gpa2hpa(vmid, virt_to_phys(&gpa2hpa));
 	if (ret < 0) {
 		pr_err("vhm: failed to inject!\n");
-- 
2.7.4

